slot.h

```diff
@@ -287,7 +291,20 @@ extern std::vector<Slot *> slots;
 extern std::vector<VSlot *> vslots;
 extern std::vector<int *> editingvslots;
 
+/**
+ * @brief Extract or replace the shader parameter name, inserting if it doesn't exist, in `shaderparams`.
+ * 
+ * Extracts the name from a global `shaderparams` whose type is `std::unordered_set<std::string>`.
+ * It performs the necessary "conversion" from `std::string` to `const char *`.
+ * 
+ * If the name doesn't exist yet, it is inserted, then returned
```

Comment:
If the name doesn't exist yet or `insert` is `true` then the value will be inserted.

It appears that due to the specific behavior of `unordered_set<std::string>` compared to `hashset<const char *>` that this may be redundant, since strings which compare equal under `strcmp` or `std::string::operator==` can be allocated independent entries in a `hashset<const char *>` (or `unordered_set<const char *>`) but not in an `unordered_set<std::string>`. I'd need to do testing to verify this behavior.